funsor/torch.py

```diff
@@ -418,7 +418,7 @@ def eager_getitem_tensor_tensor(op, lhs, rhs):
         rhs_data = rhs_data.reshape(rhs_data.shape + (1,) * (len(lhs.output.shape) - 1))
 
     # Perform advanced indexing.
-    target_dim = len(lhs.inputs) + op.offset
+    target_dim = lhs_data.dim() - len(lhs.output.shape) + op.offset
```

Comment:
This fixes a bug whereby `lhs.inputs` is no longer valid after `align_tensors()` above.